src/request.test.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import MockAdapter from "axios-mock-adapter";
+
+import { SkynetClient } from ".";
+import { ExecuteRequestError } from "./request";
+import { defaultSkynetPortalUrl } from "./utils/url";
+
+const portalUrl = defaultSkynetPortalUrl;
+const client = new SkynetClient(portalUrl);
+const skylink = "XABvi7JtJbQSMAcDwnUnmp2FKDPjg8_tTTFP4BwMSxVdEg";
+
+describe("ExecuteRequestError", () => {
+  let mock: MockAdapter;
+
+  beforeEach(() => {
+    mock = new MockAdapter(axios);
+    mock.resetHistory();
+  });
+
+  it("should have a working instanceof and also be an axios error", async () => {
+    try {
+      // Should result in a 404 as we haven't set up any mock handlers.
+      await client.getFileContent(skylink);
+      throw new Error("'getFileContent' should not have succeeded");
```

Comment:
this can be improved, you can do `expect(true).toBe(false);` but a better alternative is to list the number of asserts you expect at the beginning if your test case, like this: `expect.assertions(3);`

Reply:
I like the first option better, makes it more clear that that branch should fail and is more future-proof.